libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -35,6 +39,20 @@ export class BookSearchComponent implements OnInit {
     this.store.select(getAllBooks).subscribe(books => {
       this.books = books;
     });
+
+    this.instantSearchOnValueChange();  
+  }
+
+  instantSearchOnValueChange(){
+    this.searchForm.get('term').valueChanges.pipe(
+      takeUntil(this.unsubscriber),
+      debounceTime(500),
+      distinctUntilChanged(),
+      filter(data => data !== '')
```

Comment:
When your local is up please test this... I have a little doubt on the condition of this code. Also you might be able to avoid the filter altogether and put the condition inside distinctUntilChanged